docs/android/linking.md

```diff
@@ -0,0 +1,155 @@
+<ItemGroup>
+  <TrimmerRootAssembly Include="MyAssembly" />
+</ItemGroup>
+```
+
+> [!NOTE]
+> The `.dll` extension isn't required when setting the `TrimmerRootAssembly` MSBuild property.
+
+If the linker skips an assembly, it's considered *rooted*, which means that it and all of its statically understood dependencies will be kept. You can skip additional assemblies by adding more `TrimmerRootAssembly` MSBuild properties to the `<ItemGroup>`.
+
+## Preserve assemblies, types, and members
+
+You can pass the linker an XML description file that specifies which assemblies, types, and members need to be retained.
+
+To exclude a member from the linking process when linking all assemblies, set the `TrimmerRootDescriptor` MSBuild property in an `<ItemGroup>` tag in the project file to the XML file that defines the members to exclude:
+
+```xml
+<ItemGroup>
+  <TrimmerRootDescriptor Include="MyRoots.xml" />
+</ItemGroup>
+```
+
+The XML file then uses the trimmer [descriptor format](https://github.com/dotnet/linker/blob/main/docs/data-formats.md#descriptor-format) to define which members to exclude from linking:
+
+```xml
+<linker>
+  <assembly fullname="MyAssembly">
+    <type fullname="MyAssembly.MyClass">
+      <method name="DynamicallyAccessedMethod" />
+    </type>
+  </assembly>
+</linker>
+```
+
+In this example, the XML file specifies a method that's dynamically accessed by the app, which is excluded from linking.
+
+When an assembly, type, or member is listed in the XML, the default action is preservation, which means that regardless of whether the linker thinks it's used or not, it's preserved in the output.
+
+> [!NOTE]
+> The preservation tags are ambiguously inclusive. If you don’t provide the next level of detail, it will include all the children. If an assembly is listed without any types, then all the assembly’s types and members will be preserved.
+
+## Mark an assembly as linker safe
+
+If you have a library in your project, or you're a developer of a reusable library and you want the linker to treat your assembly as linkable, you can mark the assembly as linker safe with the [`AssemblyMetadata`](xref:System.Reflection.AssemblyMetadataAttribute) attribute:
+
+```csharp
+[assembly: AssemblyMetadata("IsTrimmable", "True")]
+```
+
+Alternatively, you can set `<IsTrimmable>true</IsTrimmable>` in a `<PropertyGroup>` tag in the project file for the assembly. This marks your assembly as "trimmable" and enable trim warnings for that project. Being "trimmable" means your assembly is considered compatible with trimming and should have no trim warnings when the assembly is built. When used in a trimmed app, the assembly's unused members will be removed in the final output.
+
+> [!NOTE]
+> If the `IsTrimmable` MSBuild property is set for an assembly, this overrides the `IsTrimmable` attribute. This enables you to opt an assembly into trimming even if it doesn't have the attribute, or to disable trimming of an assembly that has the attribute.
```

Comment:
The MSBuild property basically just puts this in the assembly for you:
```csharp
[assembly: AssemblyMetadata("IsTrimmable", "True")]
```

So maybe could rephrase this to mention it?